packages/cogify/src/cogify/cli/cli.topo.ts

```diff
@@ -0,0 +1,207 @@
+  description: 'List input topographic files, create StacItems, and generate tiles for grouping.',
+  args: {
+    ...logArguments,
+    title: option({
+      type: string,
+      long: 'title',
+      description: 'Imported imagery title',
+    }),
+    source: option({
+      type: UrlFolder,
+      long: 'source',
+      description: 'Location of the source files',
+    }),
+    target: option({
+      type: UrlFolder,
+      long: 'target',
+      description: 'Target location for the output files',
+    }),
+    scale: option({
+      type: string,
+      long: 'scale',
+      description: 'topo25, topo50, or topo250',
+    }),
+    resolution: option({
+      type: string,
+      long: 'resolution',
+      description: 'e.g. gridless_600dpi',
+    }),
+    latestOnly: flag({
+      type: boolean,
+      defaultValue: () => false,
+      long: 'latest-only',
+      description: 'Only process the latest version of each map sheet',
+      defaultValueIsSerializable: true,
+    }),
+    forceOutput: flag({
+      type: boolean,
+      defaultValue: () => false,
+      long: 'force-output',
+      defaultValueIsSerializable: true,
+    }),
+  },
+  async handler(args) {
+    const logger = getLogger(this, args);
+    const startTime = performance.now();
+    logger.info('ListJobs:Start');
+
+    const { epsgDirectoryPaths, stacItemPaths } = await loadTiffsToCreateStacs(
+      args.latestOnly,
+      args.source,
+      args.target,
+      args.title,
+      args.scale,
+      args.resolution,
+      args.forceOutput,
+      logger,
+    );
+
+    if (epsgDirectoryPaths.length === 0 || stacItemPaths.length === 0) throw new Error('No Stac items created');
+
+    // write stac items into an JSON array
+    if (args.forceOutput || isArgo()) {
+      const targetURL = isArgo() ? fsa.toUrl('/tmp/topo-stac-creation/') : args.target;
+
+      // for create-config: we need to tell create-config to create a bundled config for each epsg folder (latest only).
+      // workflow: will loop 'targets.json' and create a node for each path where each node's job is to create a bundled config.
+      await fsa.write(new URL('targets.json', targetURL), JSON.stringify(epsgDirectoryPaths, null, 2));
+
+      // tiles.json makes the tiff files
+      await fsa.write(new URL('tiles.json', targetURL), JSON.stringify(stacItemPaths, null, 2));
+      await fsa.write(new URL('brokenTiffs.json', targetURL), JSON.stringify(brokenTiffs, null, 2));
+    }
+
+    logger.info({ duration: performance.now() - startTime }, 'ListJobs:Done');
+  },
+});
+
+/**
+ * @param source: Source directory URL from which to load tiff files
+ * @example TODO
+ *
+ * @param target: Destination directory URL into which to save the STAC collection and item JSON files
+ * @example TODO
+ *
+ * @param title: The title of the collection
+ * @example "New Zealand Topo50 Map Series (Gridless)"
+ *
+ * @returns an array of StacItem objects
+ */
+async function loadTiffsToCreateStacs(
```

Comment:
this is quite a huge function and would be very easy to mix the ordering of args it would be best to use a context object here.